### ISSUE TITLE
CONCAT function does not support multibyte characters

### ISSUE THREAD

**Issue by staticlibs** [wiltondb/wiltondb#77]
[CONCAT](https://learn.microsoft.com/en-us/sql/t-sql/functions/concat-transact-sql?view=sql-server-ver16) function only works with ASCII, for example it fails with "hello1" in Bulgarian:

```tsql
select concat(N'здравейте1', N'здравейте1')
```
```
?????????1?????????1
```

**Comment by staticlibs** [comment]
The [upstream fix to this](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2949) is effective, but it causes a number of side-effect failures and is not feasible to backport.

The workaround is to implement a user-defined `CONCAT` replacement like this:

```tsql
create function my_concat(
    @s1 nvarchar(max),
    @s2 nvarchar(max),
    @s3 nvarchar(max) = N'',
    @s4 nvarchar(max) = N'',
    @s5 nvarchar(max) = N'',
    @s6 nvarchar(max) = N'',
    @s7 nvarchar(max) = N'',
    @s8 nvarchar(max) = N''
) returns nvarchar(max) as
begin
    return @s1 + @s2 + @s3 + @s4 + @s5 + @s6 + @s7 + @s8
end
```
```tsql
select my_concat(N'здравейте1', N'здравейте1')
```
```
здравейте1здравейте1
```

**Comment by staticlibs** [comment]
This is fixes in [12.16.1](https://github.com/wiltondb/wiltondb/releases/tag/3-lts-12-16-1) update.
